test/client.test.js

```diff
@@ -428,7 +431,10 @@ describe('ServiceClient', () => {
         let numberOfRetries = 0;
         clientOptions.retryOptions = {
             retries: 1,
-            onRetry() {
+            onRetry(currentAttempt, err, req) {
+                assert(currentAttempt);
+                assert(err);
```

Comment:
That just tests that you get a truthy argument. So for example if your code would accidentally swap `err` and `req` tests will still pass. Would be better to test actual values.

Reply:
Good point, updated :)